sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -275,8 +276,37 @@ object LogicalPlanIntegrity {
    * Some plan transformers (e.g., `RemoveNoopOperators`) rewrite logical
    * plans based on this assumption.
    */
-  def checkIfExprIdsAreGloballyUnique(plan: LogicalPlan): Boolean = {
-    checkIfSameExprIdNotReused(plan) && hasUniqueExprIdsForOutput(plan)
+  def validateExprIdUniqueness(plan: LogicalPlan): Unit = {
+    if (!LogicalPlanIntegrity.checkIfSameExprIdNotReused(plan)) {
+      throw SparkException.internalError("Cannot reuse the exprId in Alias")
```

Comment:
```suggestion
      throw SparkException.internalError("One or more aliases reuse the same expression ID as previously present in attribute references, which is invalid")
```

also can this mention the names, data types, and expression IDs of the aliases and attributes to help with future potential debugging?